File: Simulation/1646.Get Maximum in Generated Array/Solution.java
class Solution {
    public int getMaximumGenerated(int n) {
        int[] arr = new int[n + 1];
        int ans = 0;
        for (int i = 0; i < n + 1; i++) {
            if (i == 0 || i == 1) {
                arr[i] = i;
            }
            if (2 <= 2 * i && 2 * i <= n) {
                arr[2 * i] = arr[i];
            }
            if (2 <= 2 * i + 1 && 2 * i + 1 <= n) {
                arr[2 * i + 1] = arr[i] + arr[i + 1];
            }
            ans = Math.max(ans, arr[i]);
        }
        return ans;
    }
}

File: Simulation/1646.Get Maximum in Generated Array/Solution.py
class Solution:
    def getMaximumGenerated(self, n: int) -> int:
        arr = [0] * (n + 1)
        ans = 0
        for i in range(n + 1):
            if i == 0 or i == 1:
                arr[i] = i
            if 2 <= 2 * i and 2 * i <= n:
                arr[2 * i] = arr[i]
            if 2 <= 2 * i + 1 and 2 * i + 1 <= n:
                arr[2 * i + 1] = arr[i] + arr[i + 1]
            ans = max(ans, arr[i])
        return ans